### ISSUE TITLE
Missing parameter in definition causes error - Confirm-ParametersMatch

### ISSUE THREAD

**Issue by anwather** [Azure/enterprise-azure-policy-as-code#104]
This bug also occurs when there are no parameters and is omitted from the json which does occur so can we handle for this or if not then add some better error messages on the Confirm-ParametersMatch call

Currently fix is to add the line into the definition being blank

```json
{
    "name": "Newly created GUID",
    "properties": {
        "displayName": "Policy Display Name",
        "policyType": "Custom",
        "mode": "All",
        "description": "Policy Description",
        "metadata": {
            "version": "1.0.0",
            "category": "Your Category"
        },
        "parameters": {},
        "policyRule": {
            "if": {
                "Insert Logic Here"
            },
            "then": {
                "effect": "Audit, Deny, Modify, etc.",
                "details": {
                    "roleDefinitionIds": [],
                    "operations": []
                }
            }
        }
    }
}
```

_Originally posted by @martyh888 in https://github.com/Azure/enterprise-azure-policy-as-code/issues/58#issuecomment-1378212432_

**Comment by anwather** [comment]
@techlake - this was marked as fixed in #58 ?

**Comment by techlake** [comment]
Solved in v6.0
